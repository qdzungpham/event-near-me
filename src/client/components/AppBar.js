import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';

const styles = {
  root: {
    flexGrow: 1
  },
  flex: {
    flexGrow: 1
  }
};

const SimpleAppBar = (props) => {
  const { classes, title } = props;
  return (
    <div className={classes.root}>
      <AppBar position="sticky">
        <Toolbar>
          <Typography className={classes.flex} variant="title" color="inherit">
            {title}
          </Typography>
          <Button href="https://github.com/qdzungpham/training-events-near-me" color="inherit">
            GITHUB
          </Button>
        </Toolbar>
      </AppBar>
    </div>
  );
};

SimpleAppBar.propTypes = {
  classes: PropTypes.object.isRequired,
  title: PropTypes.string.isRequired
};

export default withStyles(styles)(SimpleAppBar);
