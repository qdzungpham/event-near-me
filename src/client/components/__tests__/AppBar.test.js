import React from 'react';
import { shallow } from 'enzyme';
import AppBar from '../AppBar';

describe('AppBar', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<AppBar title="test" />);
    expect(wrapper).toMatchSnapshot();
  });
});
