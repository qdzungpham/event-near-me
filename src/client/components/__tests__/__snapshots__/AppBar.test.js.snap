// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(SimpleAppBar)
    title="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "flex": "SimpleAppBar-flex-2",
        "root": "SimpleAppBar-root-1",
      },
      "title": "test",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "flex": "SimpleAppBar-flex-2",
          "root": "SimpleAppBar-root-1",
        },
        "title": "test",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
